packages/story-editor/src/components/library/panes/shopping/shoppingPane.js

```diff
@@ -13,81 +13,192 @@
+      setIsMenuFocused(true);
+    }
+  }, []);
+
+  const handleFocus = () => {
+    setIsMenuFocused(false);
+  };
+
+  const { deleteElementById } = useStory(({ actions }) => ({
+    deleteElementById: actions.deleteElementById,
+  }));
 
   const { insertElement } = useLibrary((state) => ({
     insertElement: state.actions.insertElement,
   }));
 
-  const insertProduct = useCallback(() => {
-    insertElement('product', {
-      width: 25,
-      height: 25,
-      product,
-    });
-    setProduct(null);
-  }, [insertElement, product]);
+  const insertProduct = useCallback(
+    (product) => {
+      insertElement('product', {
+        width: 25,
+        height: 25,
+        product,
+      });
+
+      const PRODUCT_ADDED_TEXT = sprintf(
+        /* translators: %s: product title added to page.*/
+        __('%s added to page', 'web-stories'),
+        product.productTitle
+      );
+
+      speak(PRODUCT_ADDED_TEXT);
+    },
+    [insertElement, speak]
+  );
+
+  const deleteProduct = useCallback(
+    (product) => {
+      const element = onPageProducts.find(
+        (item) => item.productId === product.productId
+      );
+      if (element) {
+        deleteElementById({ elementId: element.elementId });
+      }
+    },
+    [deleteElementById, onPageProducts]
+  );
+
+  const handleClearInput = useCallback(() => {
+    setSearchTerm('');
+  }, [setSearchTerm]);
 
-  if (!isEnabled) {
+  if (!isShoppingIntegrationEnabled) {
     return null;
   }
 
   return (
     <Pane id={paneId} {...props}>
-      <Row>
-        <HelperText>
-          {__(
-            'Select a product to add to the page. It will be displayed as a "dot" on the page, as well as in the page attachment.',
-            'web-stories'
-          )}
-        </HelperText>
-      </Row>
-      <Row>
-        <ProductDropdown product={product} setProduct={setProduct} />
-      </Row>
-      <Button
-        variant={BUTTON_VARIANTS.RECTANGLE}
-        type={BUTTON_TYPES.SECONDARY}
-        size={BUTTON_SIZES.SMALL}
-        onClick={insertProduct}
-        disabled={!product}
+      <Section
+        data-testid="products-library-pane"
+        title={__('Products', 'web-stories')}
       >
-        {__('Insert product', 'web-stories')}
-      </Button>
+        <Row>
+          <HelperText>
+            {__(
+              'This will add products as a tappable dot on your story.',
+              'web-stories'
+            )}
+          </HelperText>
+        </Row>
+        <Row>
+          <SearchInput
+            aria-label={__('Product search', 'web-stories')}
+            inputValue={searchTerm}
+            onChange={onSearch}
+            placeholder={__('Search', 'web-stories')}
+            onKeyDown={handleInputKeyPress}
+            onFocus={handleFocus}
+            isOpen
+            clearId="clear-product-search"
+            handleClearInput={handleClearInput}
+          />
+        </Row>
+        {products && products?.length >= 1 ? (
```

Comment:
Nit, since you're using optional chaining:

```suggestion
        {products?.length > 0 ? (
```